fix: workaround to wrongly renamed migration file for keypairs table

stefandunca opened the pull request:
Restore a wrongly renamed migration file done in the merge favourite [PR #2844](https://github.com/status-im/status-go/pull/2844)
  - The wrong renaming https://github.com/status-im/status-go/pull/2844/files#diff-35a2a92571286f19a4534533cb97e7c5d27c77e0580b3064f280eee8421966f2

The renaming might block startup for clients who already run the newly renamed migration and beak the onld one

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [x] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (12)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 2fa3c716 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2857/1/) | 2022-09-14 11:19:27 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220914-111716-2fa3c7-pr2857.zip) |
| :heavy_check_mark: | 2fa3c716 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2857/1/) | 2022-09-14 11:22:00 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220914-111712-2fa3c7-pr2857.aar) |
| :heavy_check_mark: | 2fa3c716 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2857/1/) | 2022-09-14 11:22:13 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220914-111716-2fa3c7-pr2857.zip) |
| | | | | | | |
| :heavy_check_mark: | 8843dda3 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2857/2/) | 2022-09-14 11:21:05 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220914-111930-8843dd-pr2857.zip) |
| :heavy_check_mark: | 8843dda3 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2857/2/) | 2022-09-14 11:25:03 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220914-112202-8843dd-pr2857.aar) |
| :heavy_check_mark: | 8843dda3 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2857/2/) | 2022-09-14 11:26:07 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220914-112215-8843dd-pr2857.zip) |
| | | | | | | |
| :heavy_check_mark: | 51245e93 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2857/3/) | 2022-09-14 13:00:49 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220914-125847-51245e-pr2857.zip) |
| :heavy_check_mark: | 51245e93 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2857/3/) | 2022-09-14 13:01:03 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220914-125847-51245e-pr2857.zip) |
| :heavy_check_mark: | 51245e93 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2857/3/) | 2022-09-14 13:01:41 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220914-125843-51245e-pr2857.aar) |
| | | | | | | |
| :heavy_check_mark: | 585895d7 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2857/4/) | 2022-09-14 13:47:26 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220914-134520-585895-pr2857.zip) |
| :heavy_check_mark: | 585895d7 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2857/4/) | 2022-09-14 13:47:58 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220914-134523-585895-pr2857.zip) |
| :heavy_check_mark: | 585895d7 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2857/4/) | 2022-09-14 13:48:32 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220914-134520-585895-pr2857.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 436c9fa8 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2857/5/) | 2022-09-14 14:15:00 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220914-141337-436c9f-pr2857.zip) |
| :heavy_check_mark: | 436c9fa8 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2857/5/) | 2022-09-14 14:15:56 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220914-141334-436c9f-pr2857.zip) |
| :heavy_check_mark: | 436c9fa8 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2857/5/) | 2022-09-14 14:18:46 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220914-141334-436c9f-pr2857.aar) |
| | | | | | | |
| :heavy_check_mark: | e5cec868 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2857/6/) | 2022-09-16 10:35:53 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220916-103422-e5cec8-pr2857.zip) |
| :heavy_check_mark: | e5cec868 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2857/6/) | 2022-09-16 10:40:21 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220916-103418-e5cec8-pr2857.aar) |
| :heavy_check_mark: | e5cec868 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2857/6/) | 2022-09-16 10:44:54 | ~10 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220916-103422-e5cec8-pr2857.zip) |

stefandunca committed:
fix: workaround for wrong renaming of migration file for keypairs table

Restore the old, previously renamed, 1662447680_add_keypairs_table.up.sql
file while keeping the current one for those who already migrated to the
new one. The extra migration is noop and saves to keep consistency in
the user data states history.

alaibe commented:
As it is already the latest migration on develop, I am not sure to understand how it will solve any issue?

AFAIK, the migration run if there is a new one after the last one, if I am correct, this new migration will never run, correct?

cammellos commented:
> he migration run if there is a new one after the last one, if I am correct, this ne

The issue is for clients that were on the removed migration, say that you have migrations 1,2,3

We release 2
Client runs 2
We delete/rename 2 to 3

If clients install that version, they will crash, because it looks for the migration 2 on start, so you will get a file not found exception.

Newer clients will migrate to 3 directly and have no issues.

alaibe commented:
@cammellos gotcha, thanks for clarifying
